trajopt_optimizers/trajopt_sqp/src/trust_region_sqp_solver.cpp

```diff
@@ -0,0 +1,481 @@
+    results_.new_exact_merit = nlp.EvaluateCostFunction(results_.new_var_vals.data()) +
+                               results_.new_constraint_violations.dot(results_.merit_error_coeffs);
+    results_.exact_merit_improve = results_.best_exact_merit - results_.new_exact_merit;
+    results_.merit_improve_ratio = results_.exact_merit_improve / results_.approx_merit_improve;
+
+    // Print debugging info
+    if (verbose_)
+      printStepInfo();
+
+    // Call callbacks
+    succeed &= callCallbacks();
+
+    // Store variables if they improved the exact merit
+    if (results_.exact_merit_improve > 1e-5)
+    {
+      results_.best_var_vals = results_.new_var_vals;
+      results_.best_exact_merit = results_.new_exact_merit;
+      results_.best_approx_merit = results_.new_approx_merit;
+      results_.best_constraint_violations = results_.new_constraint_violations;
+      nlp_->SetVariables(results_.best_var_vals.data());
+    }
+    if (SUPER_DEBUG_MODE)
+      results_.print();
+  }
+  else
+  {
+    CONSOLE_BRIDGE_logError("Solver Failure");
+
+    //    // If primal infeasible
+    //    {
+    //      Eigen::Map<Eigen::VectorXd> primal_certificate(
+    //          qp_solver_->solver_.workspace()->delta_x, qp_problem_->num_qp_cnts_, 1);
+    //      std::cout << "\n---------------------------------------\n";
+    //      std::cout << std::scientific << "Primal Certificate (v): " << primal_certificate.transpose() << std::endl;
+
+    //      double first = qp_problem_->bounds_lower_.transpose() * primal_certificate;
+    //      double second = qp_problem_->bounds_upper_.transpose() * primal_certificate;
+
+    //      std::cout << "A.transpose() * v = 0\n"
+    //                << "l.transpose() * v = " << first << "    u.transpose() * v = " << second << std::endl;
+    //      std::cout << "l.transpose() * v + u.transpose() * v  = " << first + second << " < 0\n";
+    //      std::cout << "Bounds_lower: " << qp_problem_->bounds_lower_.transpose() << std::endl;
+    //      std::cout << "Bounds_upper: " << qp_problem_->bounds_upper_.transpose() << std::endl;
+    //      std::cout << std::fixed;
+    //      std::cout << "\n---------------------------------------\n";
+    //    }
+
+    //    // If dual infeasible
+    //    {
+    //      Eigen::Map<Eigen::VectorXd> dual_certificate(
+    //          qp_solver_->solver_.workspace()->delta_y, qp_problem_->num_qp_vars_, 1);
+    //      std::cout << "\n---------------------------------------\n";
+    //      std::cout << "Dual Certificate (x): " << dual_certificate.transpose() << std::endl;
+
+    //      std::cout << "P*x = " << (qp_problem_->hessian_ * dual_certificate).transpose() << " = 0" << std::endl;
+    //      std::cout << "q.transpose() * x = " << qp_problem_->gradient_.transpose() * dual_certificate << " < 0"
+    //                << std::endl;
+    //      std::cout << std::fixed;
+    //      std::cout << "\n---------------------------------------\n";
+    //    }
```

Comment:
Should this be commented out?

Reply:
No. It has to be commented if using the osqp_eigen master branch. Once we make sure my changes to osqp_eigen work to get information like this I can submit a PR. But in the meantime it should be uncommented.